### PR TITLE
Alerting docs: update `Configure alert rules`

### PR DESCRIPTION
**Pull request by ppcano**
We've recently updated the entire Alerting introduction to clarify and simplify alerting concepts.

The following `Configure alert rules` docs  have been updated to be consistent with the messaging, include links to updated concepts, and reflect minor changes. 

- [Configure alert rules](https://deploy-preview-grafana-97430-zb444pucvq-vp.a.run.app/docs/grafana/latest/alerting/alerting-rules/)
  - [Configure Grafana-managed alert rules](https://deploy-preview-grafana-97430-zb444pucvq-vp.a.run.app/docs/grafana/latest/alerting/alerting-rules/create-grafana-managed-rule/)
  - [Configure data source-managed alert rules](https://deploy-preview-grafana-97430-zb444pucvq-vp.a.run.app/docs/grafana/latest/alerting/alerting-rules/create-data-source-managed-rule/)

### CODE REVIEW

File: docs/sources/alerting/fundamentals/alert-rules/_index.md [hunk 1]
```diff
@@ -95,25 +95,27 @@ In this setup, the distributed architecture can provide high-availability and fa
...
+| Alert rule evaluation and delivery                                                                                      | Alert evaluation is done in Grafana, while delivery can be handled by Grafana or an external Alertmanager.        | Alert rule evaluation and alert delivery are distributed.                       |
+| Scaling                                                                                                                 | Alert rules are stored in the Grafana database, which may experience transient errors. It only scales vertically. | Alert rules are stored within the data source and allow for horizontal scaling. |
 
-## Comparison between alert rule types
+## Recording rules
+
+Similar to alert rules, recording rules are evaluated periodically. A recording rule pre-compute frequently used or computationally expensive queries, and saves the results as a new time series metric.
```
Comment (irenerl24) [inline docs/sources/alerting/fundamentals/alert-rules/_index.md:117]: Similar to alert rules, recording rules are evaluated periodically. A recording rule pre-computes frequently used or computationally expensive queries, and saves the results as a new time series metric.

File: docs/sources/alerting/alerting-rules/_index.md [hunk 2]
```diff
@@ -12,37 +12,54 @@ labels:
...
-**Query, expressions, and alert condition**
+1. Data source-managed alert rules: These can only query Prometheus-based data sources and support horizontal scaling.
 
-What are you monitoring? How are you measuring it?
+We recommend using Grafana-managed alert rules whenever possible, and opting for data source-managed alert rules when horizontal scaling is required. Refer to the [comparison table of alert rule types](ref:alert-types-comparison-table) for a more detailed overview.
 
-{{< admonition type="note" >}}
-Expressions can only be used for Grafana-managed alert rules.
-{{< /admonition >}}
+Both types of alert rules can be configured in Grafana using the **+ New alert rule** flow . For step-by-step instructions, refer to:
```
Comment (irenerl24) [inline docs/sources/alerting/alerting-rules/_index.md:57]: Both types of alert rules can be configured in Grafana using the **+ New alert rule** flow. For step-by-step instructions, refer to:

File: docs/sources/alerting/alerting-rules/create-data-source-managed-rule.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+## Before you begin
+
+Verify that you have write permission to the Prometheus, Mimir, or Loki data source. Otherwise, you cannot create or update data source-managed alert rules.
+
+### Enable the Ruler API
+
+For more information, refer to the [Mimir Ruler API](/docs/mimir/latest/references/http-api/#ruler) or [Loki Ruler API](/docs/loki/latest/api/#ruler).
+
+- **Grafana Mimir** - use the `/prometheus` prefix. The Prometheus data source supports both Grafana Mimir and Prometheus, and Grafana expects that both the [Query API](/docs/mimir/latest/operators-guide/reference-http-api/#querier--query-frontend) and [Ruler API](/docs/mimir/latest/operators-guide/reference-http-api/#ruler) are under the same URL. You cannot provide a separate URL for the Ruler API.
```
Comment (irenerl24) [inline docs/sources/alerting/alerting-rules/create-data-source-managed-rule.md:93]: - **Mimir** - use the `/prometheus` prefix. The Prometheus data source supports both Grafana Mimir and Prometheus, and Grafana expects that both the [Query API](/docs/mimir/latest/operators-guide/reference-http-api/#querier--query-frontend) and [Ruler API](/docs/mimir/latest/operators-guide/reference-http-api/#ruler) are under the same URL. You cannot provide a separate URL for the Ruler API.

File: docs/sources/alerting/alerting-rules/create-data-source-managed-rule.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+- **Grafana Loki** - The `local` rule storage type, default for the Loki data source, supports only viewing of rules. To edit rules, configure one of the other rule storage types.
```
Comment (irenerl24) [inline docs/sources/alerting/alerting-rules/create-data-source-managed-rule.md:95]: - **Loki** - The `local` rule storage type, default for the Loki data source, supports only viewing of rules. To edit rules, configure one of the other rule storage types.

Comment (irenerl24) [reply docs/sources/alerting/alerting-rules/create-data-source-managed-rule.md:95]: Remove Grafana for consistency. The recommendation is to use Grafana <PRODUCT> in the first instance (line 77) and then shorten to just <PRODUCT> from then on.

File: docs/sources/alerting/fundamentals/alert-rules/_index.md (hunk 1 above)
Comment (ppcano) [reply docs/sources/alerting/fundamentals/alert-rules/_index.md]: addressed by https://github.com/grafana/grafana/pull/97430/commits/e2cda3deb2d002cece72b767fd3af5c3d3f8da71

File: docs/sources/shared/alerts/table-configure-no-data-and-error.md [hunk 5]
```diff
@@ -0,0 +1,14 @@
+---
+labels:
+  products:
+    - oss
+title: 'Table configure no data and error '
```
[the remark this answers is not in the document]
Comment (ppcano) [reply docs/sources/shared/alerts/table-configure-no-data-and-error.md]: resolved by https://github.com/grafana/grafana/pull/97430/commits/99c798423627457dd1f23876ffaa863ba8cdd53c

File: docs/sources/alerting/alerting-rules/create-data-source-managed-rule.md (hunk 3 above)
Comment (ppcano) [reply docs/sources/alerting/alerting-rules/create-data-source-managed-rule.md]: addressed by https://github.com/grafana/grafana/pull/97430/commits/da010b59b76a3de48429b7657b35f1e181c46822

**Review by brendamuir — APPROVED** [review]

**Pull request merged**